orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RecreateUpdateStrategy.groovy

```diff
@@ -0,0 +1,215 @@
+      stage,
+      SyntheticStageOwner.STAGE_BEFORE
+    )
+
+    targetPercentages.forEach({ p ->
+      def source
+      try {
+        source = getSource(targetServerGroupResolver, stageData, baseContext)
+      } catch (Exception e) {
+        stages = []
+        return stages
+      }
+      def resizeContext = baseContext + [
+        target        : TargetServerGroup.Params.Target.current_asg_dynamic,
+        action        : ResizeStrategy.ResizeAction.scale_exact,
+        source        : source,
+        targetLocation: cleanupConfig.location,
+        kind          : stageData.kind,
+        targetSize    : 0,
+        capacity      : stage.context.capacity,
+        resizeType    : "incr",
+        scaleNum      : 0
+      ]
+
+      def resizeStage = newStage(
+        stage.execution,
+        resizeServerGroupStage.type,
+        "Scale down to $p% Desired Size",
+        resizeContext,
+        stage,
+        SyntheticStageOwner.STAGE_BEFORE
+      )
+      stages << resizeStage
+
+      // an expression to grab newly deployed server group at runtime (ie. the server group being resized up)
+      def deployedServerGroupName = '${' + "#stage('${resizeStage.id}')['context']['asgName']" + '}'.toString()
+      stages.addAll(getBeforeCleanupStages(stage, stageData, source, deployedServerGroupName, p))
+    })
+
+    return stages
+  }
+
+  List<Stage> getBeforeCleanupStages(Stage parentStage,
+                                     RecreateUpdateStrategyData stageData,
+                                     ResizeStrategy.Source source,
+                                     String deployedServerGroupName,
+                                     int percentageComplete) {
+    def stages = []
+
+    def waitContext = [waitTime: stageData.delayBeforeCleanup]
+    stages << newStage(
+      parentStage.execution,
+      waitStage.type,
+      "wait",
+      waitContext,
+      parentStage,
+      SyntheticStageOwner.STAGE_BEFORE
+    )
+
+    if (stageData.pipelineBeforeCleanup?.application && stageData.pipelineBeforeCleanup?.pipelineId) {
+      def serverGroupCoordinates = [
+        region         : source.region,
+        serverGroupName: source.serverGroupName,
+        account        : source.credentials,
+        cloudProvider  : source.cloudProvider
+      ]
+
+      def pipelineContext = [
+        pipelineApplication: stageData.pipelineBeforeCleanup.application,
+        pipelineId         : stageData.pipelineBeforeCleanup.pipelineId,
+        pipelineParameters : [
+          "deployedServerGroup": serverGroupCoordinates + [
+            serverGroupName: deployedServerGroupName
+          ],
+          "sourceServerGroup"  : serverGroupCoordinates + [
+            serverGroupName: source.serverGroupName
+          ],
+          "percentageComplete" : percentageComplete
+        ]
+      ]
+
+      stages << newStage(
+        parentStage.execution,
+        pipelineStage.type,
+        "Run Validation Pipeline",
```

Comment:
Does this need a validation pipeline?

Reply:
I guess not, but I don't understand what is the purpose for validate pipeline and when I need to call it.  Can you explain?